### PR TITLE
[Misc][Bug] Add `isBatonPassable` property to `BattlerTag`s

### PR DESCRIPTION
**Pull request by DayKev**
## What are the changes?
Baton Pass is now more accurate in which properties it passes to other Pokémon.

## Why am I doing these changes?
Baton Pass currently passes `BattlerTag`s that it shouldn't to other Pokémon.

## What did change?
An `isTransferrable: boolean` property is added to `BattlerTag`s to be used by Baton Pass and the Baton item, and existing `BattlerTag`s that should be passed are marked accordingly.

## How to test the changes?
Have the opposing Pokémon apply the "infatuated" effect (or Salt Cure, or various other effects that shouldn't be passable) on your Pokémon, then use Baton Pass and verify that the effect was not passed.
`npm run test baton_pass`

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- ~[ ] Have I tested the changes (manually)?~
    - [x] Are all unit tests still passing? (`npm run test`)
- ~[ ] Are the changes visual?~
  - ~[ ] Have I provided screenshots/videos of the changes?~

### CODE REVIEW

**Review by innerthunder — APPROVED** [review]
lgtm, checked the values for each tag against https://bulbapedia.bulbagarden.net/wiki/Baton_Pass_(move)#Effect

**Review by Tempo-anon — APPROVED** [review]

File: src/data/battler-tags.ts [hunk 1]
```diff
@@ -97,7 +99,7 @@ export interface TerrainBattlerTag {
  */
 export class RechargingTag extends BattlerTag {
   constructor(sourceMove: Moves) {
-    super(BattlerTagType.RECHARGING, [ BattlerTagLapseType.PRE_MOVE, BattlerTagLapseType.TURN_END ], 2, sourceMove);
+    super(BattlerTagType.RECHARGING, [ BattlerTagLapseType.PRE_MOVE, BattlerTagLapseType.TURN_END ], 2, sourceMove, undefined, false);
```
Comment (torranx) [inline src/data/battler-tags.ts:102]: Do you need to set it to false if the default value is already `false`?

Comment (DayKev) [reply src/data/battler-tags.ts:102]: I set them all before I'd decided what the default would be lol. I can go back and remove the `false` values if needed.

Comment (torranx) [reply src/data/battler-tags.ts:102]: > I can go back and remove the `false` values if needed.

yes please

**Review by torranx — APPROVED** [review]

**Comment by Tempo-anon** [comment]
Looks good. Can you update the flash fire test mentioned in https://github.com/pagefaultgames/pokerogue/pull/3458?

**Review by innerthunder — DISMISSED** [review]

File: src/data/battler-tags.ts [hunk 2]
```diff
@@ -787,7 +789,7 @@ export class AquaRingTag extends BattlerTag {
 /** Tag used to allow moves that interact with {@link Moves.MINIMIZE} to function */
 export class MinimizeTag extends BattlerTag {
   constructor() {
-    super(BattlerTagType.MINIMIZED, BattlerTagLapseType.TURN_END, 1, Moves.MINIMIZE, undefined);
+    super(BattlerTagType.MINIMIZED, BattlerTagLapseType.TURN_END, 1, Moves.MINIMIZE);
```
Comment (Xavion3) [inline src/data/battler-tags.ts:792]: Bulbapedia contradicts itself on whether minimize can be passed with baton pass so this will need another source.

**Review by Xavion3 — DISMISSED** [review]
In addition to the above, I'm not seeing nightmare or lock on being transferrable here.

File: src/data/battler-tags.ts (hunk 2 above)
Comment (DayKev) [reply src/data/battler-tags.ts:792]: No it doesn't? Both the https://bulbapedia.bulbagarden.net/wiki/Minimize_(move) page and https://bulbapedia.bulbagarden.net/wiki/Baton_Pass_(move)#Effect page say Minimize's extra effects aren't passed.

**Review by torranx — DISMISSED** [review]

**Comment by DayKev** [comment]
> In addition to the above, I'm not seeing nightmare or lock on being transferrable here.

Bulbapedia indicates Lock On is not able to be Baton Passed in Gen5+

As for Nightmare, you can't even Baton Pass if you're affected by Nightmare, at least not in the scenario where you try to Sleep Talk into Baton Pass while affected by Nightmare (which is the only scenario you can possibly use it in).

File: src/data/battler-tags.ts (hunk 2 above)
Comment (Xavion3) [reply src/data/battler-tags.ts:792]: It does on the status effects page https://bulbapedia.bulbagarden.net/wiki/Status_condition#Minimize where it explicitly calls out baton passing minimize as a way to take the boosted damage.

**Comment by Xavion3** [comment]
Why can't you baton pass while asleep using sleep talk?

File: src/data/battler-tags.ts (hunk 2 above)
Comment (DayKev) [reply src/data/battler-tags.ts:792]: No double damage after baton pass:
image

Double damage without baton pass:
image

**Comment by DayKev** [comment]
> Why can't you baton pass while asleep using sleep talk?

https://github.com/user-attachments/assets/9ee9a995-19e1-4457-a615-96a1c0bdfaeb

**Comment by Xavion3** [comment]
If bulbapedia is this unreliable about what is transferrable should we be using it as a source at all as this did? Or should we be checking everything against smogon to decide what to do?

**Comment by DayKev** [comment]
https://www.smogon.com/dex/sv/moves/baton-pass/

Are we supposed to check every single move manually on Smogon to see if it works with Baton Pass? That's not practical. No source is going to be perfect, and to expect perfection is unreasonable.

**Comment by DayKev** [comment]
Bulbapedia also doesn't list it as transferrable actually, so I'm not sure why I marked it as such.

**Review by innerthunder — APPROVED** [review]

File: src/data/battler-tags.ts [hunk 3]
```diff
@@ -39,13 +39,15 @@ export class BattlerTag {
   public turnCount: number;
   public sourceMove: Moves;
   public sourceId?: number;
+  public isTransferrable: boolean;
```
Comment (Tempo-anon) [inline src/data/battler-tags.ts:42]: now that I think about it, can this variable be named like `isBatonPassable` to make it clear that this is just for baton pass and the baton item? isTransferrable makes me think that it's a boolean on whether or not the battler tag can be transferred at all

Comment (DayKev) [reply src/data/battler-tags.ts]: Hm, I suppose so.

**Review by Tempo-anon — APPROVED** [review]

**Review by xsn34kzx — APPROVED** [review]

**Review by frutescens — APPROVED** [review]

**Pull request merged**